src/background.js

```diff
@@ -28,6 +33,32 @@ browser.browserAction.onClicked.addListener(() => {
     openOverview()
 })
 
+
+browser.bookmarks.onCreated.addListener((id, bookmarkInfo) => {
+    //TO-DO Add method to handle indexing of bookmark
+    //bookmarkInfo.url gives the URL of the bookmark
+    //id is the unique id of the bookmark
+    //https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/bookmarks/onCreated
+    //Check if there already exists a page with this URL.
+    const samePageCandidates = (await findPagesByUrl({url})).rows.map(row => row.doc);
+    const reusablePage = await tryReidentifyPage({null, bookmarkInfo.url, samePageCandidates})
+    const pageData = fetchPageData(bookmarkInfo.url)
```

Comment:
`fetchPageData` requires an object with keys `url` and `opts` flags specifying which data to attempt to fetch. 

Also use `await` as it's an asynchronous function (`Promise` returning function), so you'll need to wait for the result before continuing on. Without `await`, the call to `fetchPageData` will return and set `pageData` to a `Promise` object and continue on immediately.

When you use `await`, you can surround the call with `try { } catch (err) { } finally { }` to handle any errors thrown during the Promise's life, which `fetchPageData` takes advantage of as there's many cases where certain data can't be fetched from a URL. In the case of a fetch error, a minimal page doc can still be made.